spec/models/waste_carriers_engine/new_registration_workflow/company_name_form_spec.rb

```diff
@@ -19,13 +19,31 @@ module WasteCarriersEngine
         end
 
         context "on back" do
-          context "when the business requires a registration number" do
-            subject { build(:new_registration, workflow_state: "company_name_form", business_type: "limitedCompany") }
+          context "when the business does not requires a registration number and the registration is upper tier" do
```

Comment:
Any chance we could use these as nested contexts? I think it would be easier to read, and also clearer what the "default" is (in this case, upper tier and a business type that requires a reg number, but right now none of that is explicit).

eg

```
context "when the registration is upper tier" do
  context "when the registration does not require a registration number" do
    ...
  end
  context "when the registration requires a registration number" do
    ... # I thought we were missing this one at first, but then realised it was at the bottom!
  end
end
context "when the registration is lower tier" do
  ...
end
```

Reply:
We don't have a case for `Upper tier` and `Registration number required`. I mean we do, but that's the default behaviour which we don't enforce with `if` or`unless`, hence I don't have a context for it.